x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -200,6 +206,34 @@ export function InnerWorkspacePanel({
     }
   }, [expression, localState.expressionBuildError]);
 
+  useEffect(() => {
+    if (
+      !activeDatasourceId ||
+      !visualizeTriggerFieldContext ||
+      localState.visualizeTriggerSuggestionsLoaded ||
+      title
+    ) {
+      return;
+    }
+
+    const suggestions = getSuggestions({
+      datasourceMap: { [activeDatasourceId]: datasourceMap[activeDatasourceId] },
+      datasourceStates,
+      visualizationMap,
+      activeVisualizationId,
+      visualizationState,
+      visualizeTriggerFieldContext,
+    });
+
+    if (suggestions.length) {
+      const selectedSuggestion =
+        suggestions.find((s) => s.visualizationId === activeVisualizationId) || suggestions[0];
+      switchToSuggestion(dispatch, selectedSuggestion, 'SWITCH_VISUALIZATION');
+      setLocalState((s) => ({ ...s, visualizeTriggerSuggestionsLoaded: true }));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Are you sure you need `useEffect` at all here? If I'm following the flow correctly, you could do this in each render:

1. Check if there is a trigger, and `!triggerSuggestionsLoaded`
2. Always set `triggerSuggestionsLoaded: true` even if the results are false
3. If there is a valid suggestion, apply it

I'm mostly saying this because we should not be disabling these linting rules, and I think the logic could be simpler this way.

Reply:
I can avoid useEffect, you are right, thanks for that :)